Modules/_testcapimodule.c

```diff
@@ -3990,6 +3990,21 @@ test_structseq_newtype_doesnt_leak(PyObject *Py_UNUSED(self),
     Py_RETURN_NONE;
 }
 
+static PyType_Spec HeapDocCType_spec;
+
+static PyObject *
+test_PyType_FromSpec(PyObject *Py_UNUSED(self), PyObject *Py_UNUSED(args))
+{
+    void *tp_doc = HeapDocCType_spec.slots[0].pfunc;
+    HeapDocCType_spec.slots[0].pfunc = NULL;
```

Comment:
Please don't modify the spec, or it will have a side effect on other tests. Instead, add a new spec with tp_doc=NULL.

I would prefer to expose such type in the _testcapi module to write a similar similar to:

```
    def test_heap_ctype_doc_and_text_signature(self):
        self.assertEqual(_testcapi.HeapDocCType.__doc__, "somedoc")
        self.assertEqual(_testcapi.HeapDocCType.__text_signature__, "(arg1, arg2)")
```

Reply:
Oh, I try to use `HeapDocCType_spec`  to avoid duplicate type_spec in here and use a temp `tp_doc` pointer to store and recover the actual  `tp_doc` value.

Updated.